src/control/lib/control/pool.go

```diff
@@ -577,6 +592,58 @@ func PoolQuery(ctx context.Context, rpcClient UnaryInvoker, req *PoolQueryReq) (
 	return pqr, convertMSResponse(ur, pqr)
 }
 
+func (smt StorageMediaType) MarshalJSON() ([]byte, error) {
+	typeStr, ok := mgmtpb.StorageMediaType_name[int32(smt)]
+	if !ok {
+		return nil, errors.Errorf("invalid storage media type %d", smt)
+	}
+	return []byte(`"` + strings.ToLower(typeStr) + `"`), nil
+}
+
+func (smt StorageMediaType) String() string {
+	return strings.ToLower(mgmtpb.StorageMediaType_name[int32(smt)])
```

Comment:
Should also be checking "ok" for these String functions. If smt is an out-of-range value, it will panic.

Reply:
done and added some unit test cases